libsodium-sys/src/crypto_generichash.rs

```diff
@@ -8,6 +8,9 @@ pub const crypto_generichash_KEYBYTES_MAX: usize = crypto_generichash_blake2b_KE
 pub const crypto_generichash_KEYBYTES: usize = crypto_generichash_blake2b_KEYBYTES;
 pub const crypto_generichash_PRIMITIVE: &'static str = "blake2b";
 
+#[allow(non_camel_case_types)]
+pub enum crypto_generichash_state { }
+
```

Comment:
I think we probably need to add the real struct definition here, please take a look at `crypto_hash_sha256.rs` to see how that's done...